astropy/io/votable/tree.py

```diff
@@ -2939,10 +2939,10 @@ def _get_binary_data_stream(self, iterator, config):
             string_io.seek(0)
             read = string_io.read
         else:
-            if not href.startswith(("http", "ftp", "file")):
+            vo_prot = ("http", "https", "ftp", "file")
+            if not href.startswith(vo_prot):
```

Comment:
This still does not check if it exactly matches the pattern _"vo_prot://..."_, but since that would technically change error handling, it's OK to follow this up in #15194.